src/rpc/rawtransaction.cpp

```diff
@@ -114,6 +114,38 @@ UniValue TxJoinSplitToJSON(const CTransaction& tx) {
     return vjoinsplit;
 }
 
+UniValue TxShieldedSpendsToJSON(const CTransaction& tx) {
+    UniValue vdesc(UniValue::VARR);
+    for (unsigned int i = 0; i < tx.vShieldedSpend.size(); i++) {
+        const SpendDescription& spendDesc = tx.vShieldedSpend[i];
+        UniValue obj(UniValue::VOBJ);
+        obj.push_back(Pair("cv", spendDesc.cv.GetHex()));
+        obj.push_back(Pair("anchor", spendDesc.anchor.GetHex()));
+        obj.push_back(Pair("nullifier", spendDesc.nullifier.GetHex()));
+        obj.push_back(Pair("rk", spendDesc.rk.GetHex()));
+        obj.push_back(Pair("proof", HexStr(spendDesc.zkproof.begin(), spendDesc.zkproof.end())));
+        obj.push_back(Pair("authSig", HexStr(spendDesc.spendAuthSig.begin(), spendDesc.spendAuthSig.end())));
+        vdesc.push_back(obj);
+    }
+    return vdesc;
+}
+
+UniValue TxShieldedOutputsToJSON(const CTransaction& tx) {
+    UniValue vdesc(UniValue::VARR);
+    for (unsigned int i = 0; i < tx.vShieldedOutput.size(); i++) {
+        const OutputDescription& outputDesc = tx.vShieldedOutput[i];
```

Comment:
Nit: `for (const OutputDescription& outputDesc : tx.vShieldedOutput) {`

Reply:
ok